### PR TITLE
BUILDER-187: Added an implementation of obtaining JWT access token

### PR DESCRIPTION
**Pull request by jamalkm**
* Added an implementation of obtaining JWT access token using Keycloak client library.
* Also, added various ways of overriding the required identity service configurations.

### CODE REVIEW

**Review by ehardon — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>${commons-io.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+            <version>${spring-boot.version}</version>
+            <scope>compile</scope>
+            <optional>true</optional>
```
Comment (rgauss) [inline pom.xml:118]: What was the reason for an optional dependency rather than sub-modules?

File: src/main/resources/authn-config.properties [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+alfresco.identity.service.credentials-secret=
+alfresco.identity.service.credentials-provider=null
+alfresco.identity.service.validate-token=false
+alfresco.identity.service.verify-token-audience=false
+# HttpClient connection timeout in millis
+alfresco.identity.service.client-connection-timeout=2000
+# HttpClient socket timeout in millis
+alfresco.identity.service.client-socket-timeout=2000
+
+alfresco.identity.service.realm-public-key=null
```
Comment (rgauss) [inline src/main/resources/authn-config.properties:23]: Do we have to provide all of these properties or can we rely on the Keycloak library's defaults?

**Review by rgauss — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (jamalkm) [reply pom.xml:118]: To make the developer life easier, the AuthConfigBuilder has an overloaded constructor that takes a Spring `Environment` object.    
In a Spring Env, our common practice is to override the properties in the `application.properties` file, therefore, all needed is to Autowire the Environment bean and pass that to the AuthnConfigBuilder. 
So, the `optional` dependency is needed in order to compile the code. Any client that wants to use the `Builder(Environment environment)` constructor, must have Spring as the dependency.

File: src/main/resources/authn-config.properties (hunk 2 above)
Comment (jamalkm) [reply src/main/resources/authn-config.properties:23]: These are the default values and we need them as we deserialize the Keycloak config object from these properties.  
However, we can get rid of all the properties that have `null` value.

Comment (rgauss) [reply src/main/resources/authn-config.properties:23]: The concern was more around having to maintain so many properties rather than those that are null.

**Review by rgauss — APPROVED** [review]

**Pull request merged**